pandas/core/generic.py

```diff
@@ -1084,6 +1084,10 @@ def to_hdf(self, path_or_buf, key, **kwargs):
                        / selecting subsets of the data
         append   : boolean, default False
             For Table formats, append the input data to the existing
+        data_columns : list of columns, or True, default None
+            This will create additional indexed columns for on-disk queries,
+            by default only 'index' and 'columns' are indexed. True will index
+            all columns.
```

Comment:
So, for a `Panel` I actually can not add columns to the index?
If that's true, how about then I add the docstring like so:
```python
data_columns : only applicable to DataFrames, see examples
```
and then I add the usage examples with the suggestions you made above?